src/net.cpp

```diff
@@ -1476,25 +1476,28 @@ void ThreadDNSAddressSeed()
     }
 
     const vector<CDNSSeedData> &vSeeds = Params().DNSSeeds();
+    const unsigned int nMaxIPs = 256;
     int found = 0;
 
     LogPrintf("Loading addresses from DNS seeds (could take a while)\n");
 
-    BOOST_FOREACH(const CDNSSeedData &seed, vSeeds) {
+    for(const CDNSSeedData &seed : vSeeds)
+    {
         if (HaveNameProxy()) {
             AddOneShot(seed.host);
         } else {
             vector<CNetAddr> vIPs;
             vector<CAddress> vAdd;
-            if (LookupHost(seed.host.c_str(), vIPs))
+            if (LookupHost(seed.host.c_str(), vIPs, nMaxIPs))
             {
-                BOOST_FOREACH(const CNetAddr& ip, vIPs)
+                for(const CNetAddr& ip : vIPs)
                 {
-                    int nOneDay = 24*3600;
+                    ++found;
+                    const int nThreeDays = 3*24*3600;
+                    const int nFourDays = 4*24*3600;
                     CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()));
-                    addr.nTime = GetTime() - 3*nOneDay - GetRand(4*nOneDay); // use a random age between 3 and 7 days old
+                    addr.nTime = GetTime() - nThreeDays - GetRand(nFourDays); // use a random age between 3 and 7 days old
                     vAdd.push_back(addr);
-                    found++;
                 }
```

Comment:
Sorry for being nitpicking but ... is it worth force an increment by one for each iteration when we already know that `found` will be incremented exactly by `vIPs.size()` ? 


```suggestion
                for(const CNetAddr& ip : vIPs)
                {
                    const int nThreeDays = 3*24*3600;
                    const int nFourDays = 4*24*3600;
                    CAddress addr = CAddress(CService(ip, Params().GetDefaultPort()));
                    addr.nTime = GetTime() - nThreeDays - GetRand(nFourDays); // use a random age between 3 and 7 days old
                    vAdd.push_back(addr);
                }
                found += vIPs.size();
```